src/ansys/acp/core/_tree_objects/extrusion_guide.py

```diff
@@ -0,0 +1,190 @@
+        the extrusion type is set to :attr:`ExtrusionGuideType.BY_DIRECTION`.
+    radius :
+        Controls the sphere of influence for mesh morphing.
+    depth :
+        Defines the bias of the mesh morphing.
+    use_curvature_correction :
+        Apply a curvature correction during the solid model extrusion which results in
+        a smoother extruded surface. Under certain circumstances, deactivating
+        curvature correction can lead to better extrusion results.
+
+    """
+
+    __slots__: Iterable[str] = tuple()
+
+    _COLLECTION_LABEL = "extrusion_guides"
+    _OBJECT_INFO_TYPE = extrusion_guide_pb2.ObjectInfo
+    _CREATE_REQUEST_TYPE = extrusion_guide_pb2.CreateRequest
+    _SUPPORTED_SINCE = "25.1"
+
+    def __init__(
+        self,
+        *,
+        name: str = "ExtrusionGuide",
+        active: bool = True,
+        edge_set: EdgeSet | None = None,
+        extrusion_guide_type: ExtrusionGuideType = "by_direction",
+        cad_geometry: VirtualGeometry | None = None,
+        direction: tuple[float, float, float] = (0.0, 0.0, 1.0),
+        radius: float = 0.0,
+        depth: float = 0.0,
+        use_curvature_correction: bool = False,
+    ):
+        super().__init__(name=name)
+        self.active = active
+        self.edge_set = edge_set
+        self.extrusion_guide_type = ExtrusionGuideType(extrusion_guide_type)
+        self.cad_geometry = cad_geometry
+        self.direction = direction
+        self.radius = radius
+        self.depth = depth
+        self.use_curvature_correction = use_curvature_correction
+
+    def _create_stub(self) -> extrusion_guide_pb2_grpc.ObjectServiceStub:
+        return extrusion_guide_pb2_grpc.ObjectServiceStub(self._channel)
+
+    status = grpc_data_property_read_only("properties.status", from_protobuf=status_type_from_pb)
+
+    active: ReadWriteProperty[bool, bool] = grpc_data_property("properties.active")
+
+    edge_set = grpc_link_property("properties.edge_set", allowed_types=(EdgeSet,))
+
+    extrusion_guide_type = grpc_data_property(
+        "properties.extrusion_guide_type",
+        from_protobuf=extrusion_guide_type_from_pb,
+        to_protobuf=extrusion_guide_type_to_pb,
+    )
+
+    cad_geometry = grpc_link_property("properties.cad_geometry", allowed_types=(VirtualGeometry,))
+
+    radius: ReadWriteProperty[float, float] = grpc_data_property("properties.radius")
+
+    depth: ReadWriteProperty[float, float] = grpc_data_property("properties.depth")
+
+    use_curvature_correction: ReadWriteProperty[bool, bool] = grpc_data_property(
+        "properties.use_curvature_correction"
+    )
+
+    # The extrusion guide type is not stored by the backend directly. Instead,
+    # it is derived from the property direction. Therefore, setting and getting
+    # the direction is blocked if the extrusion guide type is not `by_direction`.
+    # See Feature 1122546 in ADO. This can be removed once the backend is updated.
```

Comment:
Maybe the comment should mention that we'll have to support the old servers for a transition period?